hadoop-hdds/interface-client/src/main/proto/hdds.proto

```diff
@@ -451,3 +451,11 @@ message ContainerBalancerConfigurationProto {
     required bool shouldRun = 18;
     optional int32 nextIterationIndex = 19;
 }
+
+message TransferLeadershipRequestProto {
+    required bool isRandom = 1;
```

Comment:
Why do we need this flag? We can skip `isRandom`, if the nodeId is present then `isRandom = false` else `isRandom = true`. This will simplify some of the code.

Reply:
Yep, the flag is a bit wasting. I intended to explictly distinguish the cases with -n and -r.  Thx @kerneltime for the suggestion.